lighthouse-core/scripts/invoke-dt-analyze-trace.cjs

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+require('../../clients/devtools-entry.js');
+const ReportGenerator = require('../../report/generator/report-generator.js');
+
+
+// If invoked as CLI, we're gonna read latest-run's trace and analyze that (as desktop)
+if (require.main !== module) {
+  console.error('Must invoke as CLI');
+}
+
+/** @type {LH.Trace} */
+const trace = JSON.parse(
+  // Gather with:
+  //     lighthouse https://www.nytimes.com --preset=desktop --only-categories=performance -GA --throttling-method=devtools
+    require('fs').readFileSync(__dirname + '/../../latest-run/defaultPass.trace.json', 'utf8')
+);
+
+/**
+   * @param {LH.Trace} trace
+   */
+const getInitialUrl = trace => {
+  // TODO: this technique is wrong. it broke on the rv camping site.
+  const urls = trace.traceEvents.filter(e =>
+    (e.name === 'navigationStart' && e?.args?.data?.isLoadingMainFrame === true)
+    // || e.name === 'NavigationBodyLoader::StartLoadingBody'
+  )
+  .map(e => e.args.data?.documentLoaderURL || e.args.url)
+  .filter(Boolean);
+  // find most common item: https://stackoverflow.com/a/20762713/89484
+  return urls.sort(
```

Comment:
why isn't this just returning the first item, wouldn't that be the initial url?

Reply:
maybe devtools can just tell us what it is as part of analyzeTrace params if it's so wishy washy